### PR TITLE
Upgrade orc version

### PR DESCRIPTION
**Pull request by galipremsagar**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by nehaljwani** [comment]
@galipremsagar You'll need to backport https://github.com/noirello/pyorc/commit/af5a7cd6036520b8c4cebd259f5ca5fc2fcdbe7c as a patch to this recipe to make the build pass. And I think you should also unpin the version from the recipe and let the global pinning take care of it. That way, whenever there is an update to orc, the bot will automatically open a PR here.

**Comment by galipremsagar** [comment]
> @galipremsagar You'll need to backport [noirello/pyorc@af5a7cd](https://github.com/noirello/pyorc/commit/af5a7cd6036520b8c4cebd259f5ca5fc2fcdbe7c) as a patch to this recipe to make the build pass. And I think you should also unpin the version from the recipe and let the global pinning take care of it. That way, whenever there is an update to orc, the bot will automatically open a PR here.

Thanks, @nehaljwani Addressed your reviews.

File: recipe/patches/0001-Fix-software-version-test.patch [hunk 1]
```diff
@@ -1,3 +1,5 @@
+https://github.com/conda-forge/pyorc-feedstock/pull/18#issuecomment-1012846145
+
```
Comment (jakirkham) [inline recipe/patches/0001-Fix-software-version-test.patch:2]: Ah would suggest adding this to `meta.yaml` as a comment. Also would be worth including the link to the commit

File: recipe/patches/0001-Fix-software-version-test.patch [hunk 2]
```diff
@@ -0,0 +1,35 @@
+https://github.com/conda-forge/pyorc-feedstock/pull/18#issuecomment-1012846145
```
Comment (galipremsagar) [inline recipe/patches/0001-Fix-software-version-test.patch:1]:
```suggestion
https://github.com/noirello/pyorc/commit/af5a7cd6036520b8c4cebd259f5ca5fc2fcdbe7c
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -5,6 +5,8 @@ package:
 source:
   url: https://pypi.io/packages/source/p/pyorc/pyorc-0.5.0.tar.gz
   sha256: 89c4c708c653d0fe90f251fbcc4a6057daf7cf63c23dfe3d644d9ed2603e296d
+  patches:
+    - patches/0001-Fix-software-version-test.patch
```
Comment (galipremsagar) [inline recipe/meta.yaml:9]:
```suggestion
  patches:
    - patches/0001-Fix-software-version-test.patch   # https://github.com/conda-forge/pyorc-feedstock/pull/18#issuecomment-1012846145
```

File: recipe/patches/0001-Fix-software-version-test.patch (hunk 1 above)
Comment (galipremsagar) [reply recipe/patches/0001-Fix-software-version-test.patch]: Done 👍

File: recipe/meta.yaml [hunk 4]
```diff
@@ -5,6 +5,8 @@ package:
 source:
   url: https://pypi.io/packages/source/p/pyorc/pyorc-0.5.0.tar.gz
   sha256: 89c4c708c653d0fe90f251fbcc4a6057daf7cf63c23dfe3d644d9ed2603e296d
+  patches:
+    - patches/0001-Fix-software-version-test.patch   # https://github.com/conda-forge/pyorc-feedstock/pull/18#issuecomment-1012846145
```
Comment (jakirkham) [inline recipe/meta.yaml:9]:
```suggestion
    # https://github.com/conda-forge/pyorc-feedstock/pull/18#issuecomment-1012846145
    - patches/0001-Fix-software-version-test.patch
```

**Comment by jakirkham** [comment]
@conda-forge/pyorc, this is ready for review! 😄

**Review by noirello — APPROVED** [review]
Thank you, looks great. 

I put the pin because of that broken test. Thought I'll remove it with 0.6.0. But this works just as well. 😄

**Comment by jakirkham** [comment]
Thanks @noirello! 😄

Yeah that makes sense.

We were just discussing over in issue ( https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/2400 ) that we can now leverage some of conda-forge's automation to get new PRs for `orc` updates added automatically with this change. So hopefully that makes things a bit smoother for future `orc` releases 🙂

File: recipe/meta.yaml [hunk 5]
```diff
@@ -5,6 +5,9 @@ package:
 source:
   url: https://pypi.io/packages/source/p/pyorc/pyorc-0.5.0.tar.gz
   sha256: 89c4c708c653d0fe90f251fbcc4a6057daf7cf63c23dfe3d644d9ed2603e296d
+  patches:
+    # https://github.com/conda-forge/pyorc-feedstock/pull/18#issuecomment-1012846145
+    - patches/0001-Fix-software-version-test.patch
 
 build:
   number: 1
```
Comment (jakirkham) [inline recipe/meta.yaml:13]: We just realized we forgot to bump the build number 🤦‍♂️

PR ( https://github.com/conda-forge/pyorc-feedstock/pull/19 ) should address this

**Pull request merged**